client/src/app/queries/rulebundles.ts

```diff
@@ -0,0 +1,140 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { BundleOrderSetting, IReadFile, RuleBundle } from "@app/api/models";
+import {
+  createFile,
+  createRuleBundle,
+  deleteRuleBundle,
+  getRuleBundles,
+  getSettingById,
+  updateRuleBundle,
+} from "@app/api/rest";
+import { AxiosError } from "axios";
+
+export const RuleBundlesQueryKey = "rulebundles";
+
+export const useFetchRuleBundles = () => {
+  const { data, isLoading, error, refetch } = useQuery<RuleBundle[]>(
+    [RuleBundlesQueryKey],
+    async () => await getRuleBundles(),
+    {
+      onError: (err) => console.log(err),
+    }
+  );
+  return {
+    ruleBundles: data || [],
+    isFetching: isLoading,
```

Comment:
Ahh.. the reason the effect I mentioned above isn't looping is because we're really using the `isLoading` property from `useQuery` and not the `isFetching` property. `isLoading` will be true only during the first fetch, so that's why we don't get a loop. It does mean the effect is triggered twice though, is that intentional?

We probably want to be more explicit (fetch something only on mount with `[]` as the useEffect deps, then handle any cascading fetches as `onSuccess` or something like that. or if these things always need to be fetched in sequence, we could combine them into one `useQuery` and await one before fetching the other?) 

If not, we probably want to use the name `isLoading` instead of `isFetching` to avoid confusion with the `isFetching` from useQuery. Although if that's a pattern duplicated elsewhere, cleaning that up is probably out of scope here.